microprofile.jdt/org.eclipse.lsp4mp.jdt.test/src/main/java/org/eclipse/lsp4mp/jdt/core/snippets/JavaFileCursorContextTest.java

```diff
@@ -42,11 +46,10 @@ public class JavaFileCursorContextTest extends BasePropertiesManagerTest {
 
 	@After
 	public void cleanUp() throws Exception {
```

Comment:
I actually think this "cleanup" needs to be moved into `BasePropertiesManagerTest`. If you have 2 or more testcases that import the exact same sample project, the sample project is copied into place. The previous `${project-dir}/.settings/` folder disappears. It  gets generated only by the project import operation. However, the `.settings/` folder doesn't get regnerated because the Eclipse project model still contains the stale project. The import doesn't occur and things like project compiler settings are incorrect (ie. `JavaCore.COMPILER_COMPLIANCE` is `23` (global default) instead of `1.8` (pom configured)).

The solution is to delete the project not just on the filesystem, but in the Eclipse project model as well.

However, looking around, I noticed **many** other projects that don't do this cleanup, and they didn't seem to have any issues with compiler options. Actually, something as simple as `project.refreshLocal(IResource.INFINITE, null)` seems to trigger a refresh which exposes the problem (compiler options disappear). So we really need to future-proof this.

Reply:
I've decided to address this in a separate contribution. I know roughly which 2 or 3 tests are failing but I'd like to address it in a separate commit.